common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -314,6 +322,67 @@ public static String simpleClassName(Class<?> clazz) {
         }
     }
 
+    /**
+     * Escapes the specified value, if necessary according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     * <p>
+     * The {@link CharSequence} will be escaped is if contains a
+     * <ul>
+     *     <li>{@link StringUtil#COMMA}</li>
+     *     <li>{@link StringUtil#LINE_FEED}</li>
+     *     <li>{@link StringUtil#CARRIAGE_RETURN}</li>
+     *     <li>{@link StringUtil#DOUBLE_QUOTE}</li>
+     * </ul>
+     * If a double quote is found inside of the value, it will be escaped by preceeding it
+     * with another double quote.
+     *
+     * @param value the value to be appended, and quoted if necessary
+     * @return {@link CharSequence} the quoted value
+     */
+    public static CharSequence escapeCsv(CharSequence value) {
+        int length = checkNotNull(value, "value").length();
+        if (length == 0) {
+            return value;
+        }
+        int first = 0;
+        char previous = value.charAt(first);
+        int last = length - 1;
+        boolean quoted = isDoubleQuote(previous) && isDoubleQuote(value.charAt(last)) && length != 1;
+        boolean needsQuote = false;
+        StringBuilder escaped = new StringBuilder(DOUBLE_QUOTE_STRING);
+        for (int i = 0; i < length; i++) {
+            char current = value.charAt(i);
+            switch (current) {
+                case DOUBLE_QUOTE:
+                    // don't quote if already quoted.
+                    if ((i == first || i == last) && quoted) {
+                        continue;
+                    }
+                    // if the current char is not escaped, escape it. But only if it is not already escaped.
+                    // special case where a double quote exists before the last closing quote.
+                    if (!quoted || !isDoubleQuote(previous) &&
+                            (!isDoubleQuote(next(value, i)) || isDoubleQuote(next(value, i)) && i + 1 == last)) {
+                        escaped.append(DOUBLE_QUOTE);
+                    }
+                case LINE_FEED:
+                case CARRIAGE_RETURN:
+                case COMMA:
+                    needsQuote = true;
+            }
+            escaped.append(current);
+            previous = current;
+        }
+        return needsQuote ? escaped.append(DOUBLE_QUOTE).toString() : value;
+    }
+
+    private static boolean isDoubleQuote(char c) {
+        return c == DOUBLE_QUOTE;
+    }
+
+    private static char next(CharSequence charSequence, int index) {
```

Comment:
Instead of a method, lets store the result of `charSequence.charAt(index + 1);` in a variable and use that.  This method is used twice in the same code block.